pinot-spi/src/main/java/org/apache/pinot/spi/utils/JsonUtils.java

```diff
@@ -411,9 +419,10 @@ private static void unnestResults(List<Map<String, String>> currentResults,
   public static Schema getPinotSchemaFromJsonFile(File jsonFile,
```

Comment:
When parsing schema, we should always allow big decimal (as default value). No need to add the extra boolean for this

Reply:
As mentioned in the other comment, enabling parsing of exact BigDecimal may end up causing floats/doubles to be parsed as BigDecimal too and we need to cast to the proper data type so that code that compare parsed vs. actual does not fail.